### PR TITLE
Use simpler initialization of std::set

### PR DESCRIPTION
**Pull request by kronbichler**
As suggested by https://github.com/dealii/dealii/pull/13674#issuecomment-1118646185 - thanks @bangerth for the good hint for modernizing things.

### CODE REVIEW

**Review by bangerth — APPROVED** [review]

File: tests/matrix_free/copy_feevaluation.cc [hunk 1]
```diff
@@ -172,8 +172,7 @@ test()
   stokes_sub_blocks[dim] = 1;
   DoFRenumbering::component_wise(dof_handler, stokes_sub_blocks);
 
-  std::set<types::boundary_id> no_normal_flux_boundaries;
-  no_normal_flux_boundaries.insert(1);
+  std::set<types::boundary_id> no_normal_flux_boundaries = {1};
```
Comment (marcfehling) [inline tests/matrix_free/copy_feevaluation.cc:175]:
```suggestion
  const std::set<types::boundary_id> no_normal_flux_boundaries = {1};
```
I think `const` should work here, too.

**Review by marcfehling — APPROVED** [review]

File: tests/matrix_free/copy_feevaluation.cc (hunk 1 above)
Comment (kronbichler) [reply tests/matrix_free/copy_feevaluation.cc]: I agree. I widened my search and found some more appearances.

**Comment by marcfehling** [comment]
```
/jenkins/workspace/dealii_PR-13676/tests/dofs/dof_tools_09.cc: In function ‘void check_this(const dealii::DoFHandler<dim>&)’:
/jenkins/workspace/dealii_PR-13676/tests/dofs/dof_tools_09.cc:45:24: error: no matching function for call to ‘std::set<unsigned int>::insert(int) const’
   boundary_ids.insert(1);
                        ^
```
Another id will be added after you initialize the container as `const`.

File: examples/step-37/step-37.cc [hunk 2]
```diff
@@ -841,8 +841,7 @@ namespace Step37
     const unsigned int nlevels = triangulation.n_global_levels();
     mg_matrices.resize(0, nlevels - 1);
 
-    std::set<types::boundary_id> dirichlet_boundary;
-    dirichlet_boundary.insert(0);
+    const std::set<types::boundary_id> dirichlet_boundary = {0};
```
Comment (bangerth) [inline examples/step-37/step-37.cc:844]: Since you're already in this patch, would you mind terribly changing the name for the variable to `dirichlet_boundary_ids`? Because it's really a set of ids, not of boundaries.

Comment (kronbichler) [reply examples/step-37/step-37.cc]: Not at all - I fixed this here and in step-66.

**Comment by kronbichler** [comment]
@marcfehling I fixed this problem - and I also updated the test, because it did not actually use that function with the particular boundary id as it was supposed to be from the test description. Now it should be fine.

File: tests/dofs/dof_tools_09.cc [hunk 3]
```diff
@@ -44,7 +43,7 @@ check_this(const DoFHandler<dim> &dof_handler)
 
   // check for boundary id 0 and 1
   boundary_ids.insert(1);
-  DoFTools::map_dof_to_boundary_indices(dof_handler, map);
+  DoFTools::map_dof_to_boundary_indices(dof_handler, boundary_ids, map);
```
Comment (marcfehling) [inline tests/dofs/dof_tools_09.cc:46]: Good catch! :)

**Review by marcfehling — APPROVED** [review]
CI is happy. Ready to merge.
